cpp/src/gandiva/precompiled/time_test.cc

```diff
@@ -315,6 +316,18 @@ TEST(TestTime, TimeStampAdd) {
   EXPECT_EQ(add_date64_int64(StringToTimestamp("2000-02-27 00:00:00"), 4),
             StringToTimestamp("2000-03-02 00:00:00"));
 
+  EXPECT_EQ(
```

Comment:
I think it would be good to add a test using leap year, both for `add` and `subtract` functions, to see if it is working correctly in that case.

Reply:
there's already a test case for the add function to check if it's considering then, I modified the test that should test it on  subtract to be more effective.